models/ArticleManager.php

```diff
@@ -0,0 +1,377 @@
+                    articles.chapo,
+                    articles.image,
+                    articles.content,
+                    articles.date,
+                    admins.name
+            FROM articles
+            JOIN admins
+            ON articles.writer = admins.email
+            WHERE articles.id='{$_GET['id']}'
+            AND articles.posted = '1'
+        ");
+    
+        $post = [];
+        while($row = $req->fetch()){
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $post;
+    }
+
+
+//FONCTIONS POUR LE BACKEND
+    // Fonction permettant de récupérer l'ensemble des articles
+    function get_posts(){
+        $req = $this->db->query("SELECT * FROM articles ORDER BY date DESC");
+        
+        $results = [];
+        while($rows = $req->fetchObject()){
+            $results[] = $rows;
+        }
+        
+        return $results;
+    }
+
+// Fonction permettant d'obtenir un article en particulier
+    function get_post(){
+        $req = $this->db->query("
+            SELECT  articles.id,
+                    articles.title,
+                    articles.chapo,
+                    articles.image,
+                    articles.date,
+                    articles.content,
+                    articles.posted,
+                    admins.name
+            FROM articles
+            JOIN admins
+            ON articles.writer = admins.email
+            WHERE articles.id = '{$_GET['id']}'
+        ");
+     $post = [];
+        while($row = $req->fetch()){
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $post;
+    }
+
+    // Fonction permettant d'éditer un article 
+    function edit($title,$chapo,$content,$posted,$id){
+        $tableau = [
+            'title'     => $title,
+            'chapo'     => $chapo,
+            'content'   => $content,
+            'posted'    => $posted,
+            'id'        => $id
+        ];
+
+        $sql = "UPDATE articles SET title=:title, chapo=:chapo, content=:content, date=NOW(), posted=:posted WHERE id=:id";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+    }
+
+    // Fonction permettant d'insérer un nouvel article dans la base de données
+    function post($title, $chapo, $content,$posted){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [post accesses the super-global variable $_SESSION.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)